### DISCUSSION TITLE
Map auto filled

### DISCUSSION THREAD

**Discussion by 191brian** [nesbox/TIC-80#1459] [Q&A] [answered]
Hi 

I am new to TIC-80 I notice if I draw a sprite/tile then go the map editor it auto fills the whole map with the one sprite/tile.
Is this by design or have I done some thing wrong?
Is there a quick way to clear the whole map?

Brian ...

**Comment by joshgoebel** [comment, marked as answer]
Don't draw a sprite in position zero. ￼ the entire map is set to zero.

**Reply by 191brian** [reply]
Thank you makes sense now!
Brian ...

**Comment by The-Beesh-Spweesh** [comment]
All entries on the map will point to sprite 0 by default. That sprite should normally be left blank to be used as a fully-transparent tile as long as the color key parameter is set correctly.

**Reply by 191brian** [reply]
Thank you, how do I check the color key parameter?

**Reply by The-Beesh-Spweesh** [reply]
The map() function has a "color key" parameter ranging from 0 to 15 specifying which color is to be transparent when drawing the map, with -1 resulting in the whole map being opaque. It can also take an array of values to specify multiple transparent color keys simultaneously.
